data_utils.py

```diff
@@ -67,61 +69,65 @@ def tokenize(text):
     return [t.lower() for t in tokenizer.tokenize(text) if not t.isnumeric()]
 
 
-def _load_raw_data(path, is_test=False):
-    log.info(f'Load data from {path}.')
+def _load_raw_data(config, path, is_test=False):
+    logging.info(f'Load data from {path}.')
     data = pd.read_csv(path, sep='\t', names=['label', 'text'],
                        converters={'label': lambda s: s.split(),
                                    'text': tokenize})
     data = data.reset_index().to_dict('records')
     if not is_test:
         data = [d for d in data if len(d['label']) > 0]
+    if config.fixed_length:
+        pad_seq = [PAD] * config.max_seq_length
+        for i in range(len(data)):
+            pad_len = config.max_seq_length - len(data[i]['text'])
+            data[i]['text'] += pad_seq[:pad_len]
     return data
 
 
-@log.enter('load_dataset')
 def load_datasets(config):
     datasets = {}
     test_path = config.test_path or os.path.join(config.data_dir, 'test.txt')
     if config.eval:
-        datasets['test'] = _load_raw_data(test_path, is_test=True)
+        datasets['test'] = _load_raw_data(config, test_path, is_test=True)
     else:
         if os.path.exists(test_path):
-            datasets['test'] = _load_raw_data(test_path, is_test=True)
+            datasets['test'] = _load_raw_data(config, test_path, is_test=True)
         train_path = config.train_path or os.path.join(config.data_dir, 'train.txt')
-        datasets['train'] = _load_raw_data(train_path)
+        datasets['train'] = _load_raw_data(config, train_path)
         val_path = config.val_path or os.path.join(config.data_dir, 'valid.txt')
         if os.path.exists(val_path):
-            datasets['val'] = _load_raw_data(val_path)
+            datasets['val'] = _load_raw_data(config, val_path)
         else:
             datasets['train'], datasets['val'] = train_test_split(
                 datasets['train'], test_size=config.val_size, random_state=42)
 
     msg = ' / '.join(f'{k}: {len(v)}' for k, v in datasets.items())
-    log.info(f'Finish loading dataset ({msg})')
+    logging.info(f'Finish loading dataset ({msg})')
     return datasets
 
 
 def load_or_build_text_dict(config, dataset):
     if config.vocab_file:
-        log.info(f'Load vocab from {config.vocab_file}')
+        logging.info(f'Load vocab from {config.vocab_file}')
         with open(config.vocab_file, 'r') as fp:
-            vocab_list = ['**PAD**'] + [vocab.strip() for vocab in fp.readlines()]
-        vocabs = Vocab(collections.Counter(vocab_list), specials=['<unk>'],
+            vocab_list = [PAD] + [vocab.strip() for vocab in fp.readlines()]
```

Comment:
I suggest we add a comment here:

# Due to the tentative need to reproduce CAML results, if vocab_file is used,
# PAD is put in the beginning instead of in the end

Reply:
Why we need this comment?
For the other case (vocab_file is not used), PAD is also put in the beginning.

Reply:
Yes, you are right.. So please modify the comment to say the difference you mentioned

Reply:
OK. Done.